### ISSUE TITLE
Sheet: Focus management fallback on open

### ISSUE THREAD

**Issue by sirrah-tam** [ithaka/pharos#1040]
**Expected behavior**
When the sheet opens, there should be a consistent behavior where keyboard focus moves, either to the sheet outer container, the handle, or somewhere where on next tab stop focus should be within the sheet. 

**Actual behavior**
When there is no specified focus target, focus remains on the triggering element behind the opened sheet. Users are then required to navigate the rest of the page to get to the sheet. This is the current behavior that is seen within the Storybook example

**Steps to reproduce the issue**

1. Navigate to and activate the triggering element that opens the sheet component
2. Using the keyboard only, attempt to navigate to the sheet component
3. Observe where focus moves is located
4. Alternatively, once the sheet appears, use the browser console to inspect the current active element using `document.activeElement`

**Screenshots or code**

<img width="459" height="338" alt="screenshot of sheet component with focus indication still on triggering element" src="https://github.com/user-attachments/assets/30db6df3-6871-4f14-ad0f-534e3ffcbeaf" />

**Pharos version**
14.16.2

**Additional information**
Passing elements with the `data-sheet-focus` attribute into the sheet's slot does work as expected, but the documentation around use of this isn't clear. Additionally, if nothing is used there doesn't appear to be an applicable element in the `focusable` list that helps here.
